File: src/SmartSolar.Application/DTOs/Auth/LoginRequest.cs
/*
 * File: LoginRequest.cs
 * Project: Smart Solar Microgrid Trading System
 * Purpose: Shared project source file for the SE4040 EAD implementation.
 * Note: Keep this header and add/update method-level comments as the code evolves.
 */

using System.ComponentModel.DataAnnotations;

namespace SmartSolar.Application.DTOs.Auth;

public sealed class LoginRequest
{
    [Required]
    public string Nic { get; init; } = string.Empty;

    [Required]
    public string Password { get; init; } = string.Empty;
}

File: mobile/SmartSolarMobile/app/src/main/java/com/smartsolar/mobile/data/remote/dto/LoginRequest.java
package com.smartsolar.mobile.data.remote.dto;

public class LoginRequest {
    private final String nic;
    private final String password;

    public LoginRequest(String nic, String password) {
        this.nic = nic;
        this.password = password;
    }

    public String getNic() { return nic; }
    public String getPassword() { return password; }
}
